### ISSUE TITLE
"Can't find npm module" when running a "meteor build" bundle on Meteor 1.3.2.4

### ISSUE THREAD

**Issue by benweissmann** [meteor/meteor#7059]
After upgrading to Meteor 1.3, i'm having trouble running a bundle built with `meteor build`. After running an `npm install` in `programs/server`, i start up the app and it errors with `Can't find npm module 'multivariate-normal'. Did you forget to call 'Npm.depends' in package.js within the 'modules-runtime' package?`

I'm importing  `multivariate-normal` with `import MultivariateNormal from "multivariate-normal";` in a file in my app. This file is in my `imports` directory, so it's lazily loaded (by the modules package, i assume). I've listed `multivariate-normal` as a dependency in my `package.json`, and the app runs fine in development.

Looking at the bundle, it looks like `programs/server/boot.js` has some node module resolution logic in it to isolate the modules for each package. By adding a `console.log` to the Npm.require defined in `boot.js`, I determined that `nonLocalNodeModulesPaths` is `npm/node_modules/meteor/modules-runtime/node_modules`. This folder contains only an `install` package, not all of my app's dependencies. My app's dependencies are in the top level `npm/node_modules` folder, which is registered as a `nonLocalNodeModulesPaths` for `app.js`, but not `modules.js`, and the code from my app that needs `multivariate-normal` is ending up in `modules.js`.

I'm working around the issue by replacing

    {
      "node_modules": "npm/node_modules/meteor/modules-runtime/node_modules",
      "sourceMap": "packages/modules-runtime.js.map",
      "path": "packages/modules-runtime.js"
    },

with 

    {
      "node_modules": {"npm/node_modules/meteor/modules-runtime/node_modules": {}, "npm/node_modules": {}},
      "sourceMap": "packages/modules-runtime.js.map",
      "path": "packages/modules-runtime.js"
    },

in `programs/server/program.json`.

Any idea why the bundle isn't working out of the box?

**Comment by abernix** [comment]
Please submit a complete reproduction (with a Github repository link) for this issue or consider asking for help on [Meteor forums](http://forums.meteor.com/) or Stack Overflow.  While it sounds like you're doing things correctly, trying to piece everything together from your explanation is too tricky and prone to misunderstanding. Issues are reserved for feature requests or documented bugs.  Please read more at https://github.com/meteor/meteor/blob/devel/Contributing.md#reporting-bug

Feel free to @mention me for a re-open if you feel you have found a reproducible bug and you have a sample reproduction.

**Comment by pierreozoux** [comment]
arf same issue here:

try it here:
https://github.com/pierreozoux/TioApp/

You can try with my Dockerfile.
I get this when I `docker-compose up`:

```
app_1 | 
app_1 | /app/bundle/programs/server/node_modules/fibers/future.js:280
app_1 | 						throw(ex);
app_1 | 						      ^
app_1 | Error: Can't find npm module 'twilio'. Did you forget to call 'Npm.depends' in package.js within the 'modules-runtime' package?
app_1 |     at Object.Npm.require (/app/bundle/programs/server/boot.js:203:17)
app_1 |     at options.fallback (packages/modules-runtime/modules-runtime.js:21:1)
app_1 |     at Module.require (packages/modules-runtime/.npm/package/node_modules/install/install.js:88:1)
app_1 |     at Module.Mp.import (/app/bundle/programs/server/npm/node_modules/meteor/modules/node_modules/reify/lib/runtime.js:70:16)
app_1 |     at meteorInstall.server.lib.sms.js (server/lib/sms.js:1:20)
app_1 |     at fileEvaluate (packages/modules-runtime/.npm/package/node_modules/install/install.js:153:1)
app_1 |     at require (packages/modules-runtime/.npm/package/node_modules/install/install.js:82:1)
app_1 |     at i18n/pt.i18n.yml:20:1
app_1 |     at /app/bundle/programs/server/boot.js:297:10
app_1 |     at Array.forEach (native)
```

**Comment by abernix** [comment]
@pierreozoux I'd expect you're getting a `Unable to resolve some modules` warning about this when you run `meteor build --directory /app` from your `Dockerfile`, but you should be running `meteor npm install` at least one in your project root _before_ you run `meteor build`.  In development you'd have already done this, but if you're building from a CI server or `Dockerfile`, you'd need to do it there too since your `node_modules` aren't checked into Git (which is correct & fine).

Basically, add `meteor npm install` **before** [this line](https://github.com/pierreozoux/TioApp/blob/master/Dockerfile#L10) in your `Dockerfile`.

**Comment by pierreozoux** [comment]
Thanks a lot @abernix you saved my day!

I think I was getting mad because of this warning I was seeing before:
```
Unable to resolve some modules:

  "twilio" in /source/server/lib/sms.js (os.linux.x86_64)

If you notice problems related to these missing modules, consider running:

  meteor npm install --save twilio
```

I think it should be helpful to either have another warning, or run `meteor npm install` automatically before `meteor` or `meteor build`.

What do you think, should I open an issue about it?

**Comment by abernix** [comment]
@pierreozoux Well, if you ran `meteor npm install --save twilio` that would have fixed it too :smile:, however there is already #6848 open for the general `npm install` automatic-ness if you want to +1 it!  Glad it fixed it for you!

**Comment by pierreozoux** [comment]
I did :) But, yeah, I guess #6848 will smoothen things :)
